src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -36,57 +36,37 @@ private static EqualityComparer<T> CreateComparer()
             Contract.Ensures(Contract.Result<EqualityComparer<T>>() != null);
             
             object result = null;
-            RuntimeType t = (RuntimeType)typeof(T);
+            var type = (RuntimeType)typeof(T);
             
-            // Specialize type byte for performance reasons
-            if (t == typeof(byte)) {
+            // Specialize for byte so Array.IndexOf is faster.
+            if (type == typeof(byte))
```

Comment:
`CreateInstanceForAnotherGenericParameter` creates the instance for you. `Activator.CreateInstance` should not be necessary. I think that the path of last resort that creates `ObjectComparer` should look exactly same as all the other more optimized generic paths.